src/lib/mac/x919_mac/x919_mac.cpp

```diff
@@ -30,12 +30,12 @@ void ANSI_X919_MAC::add_data(std::span<const uint8_t> input) {
 
    m_des1->encrypt(m_state);
    while(in.remaining() >= 8) {
-      xor_buf(m_state, in.take(8).data(), 8);
+      xor_buf(std::span{m_state}.first<8>(), in.take<8>());
```

Comment:
Just for the record: I just realized that `m_state` is actually 8 bytes wide by definition. Technically, this should work too:

```suggestion
      xor_buf(m_state, in.take<8>());
```

... BUT this change would incur a runtime length check for every iteration, because passing a `secure_vector` does not carry the compile-time length information.

Reply:
You are right, if we make this change, the length check will be repeated every iteration due to typecast_copy. We made some changes accordingly in the previous review.

Such unnecessary runtime checks can lead to performance loss and conflict with the goal of optimized readable code. Therefore, we could push this check to compile time by defining a length constant constexpr and using a fixed-size span like std::span<uint8_t, N>.

However, this solution may slightly reduce the readability of the code. There is an expression in Turkish. “The stone you throw is not worth the bird you scare.” (An idiom used when the resources used to do a job are more than the benefit you will get as a result of the job). Consider the following example:

constexpr size_t STATE_SIZE = 8;
xor_buf(std::span<uint8_t, STATE_SIZE>(m_state.data(), STATE_SIZE), in.take<STATE_SIZE>());

I am not very sure about readability; I think this point should be taken into account if changes are made.

Reply:
Yeah, I agree. This can and should be tackled when we explicitly refactor this class. Fine by me to leave as-is (i.e. with the magic numbers) for now.